core/src/main/java/com/linecorp/armeria/server/healthcheck/CpuHealthChecker.java

```diff
@@ -0,0 +1,174 @@
+    static final MethodHandle systemCpuUsage;
+
+    @Nullable
+    @VisibleForTesting
+    static final MethodHandle processCpuUsage;
+
+    static {
+        operatingSystemBeanClass = requireNonNull(getFirstClassFound(OPERATING_SYSTEM_BEAN_CLASS_NAMES));
+        operatingSystemBean = ManagementFactory.getOperatingSystemMXBean();
+        final MethodHandle getCpuLoad = detectMethod("getCpuLoad");
+        systemCpuUsage = getCpuLoad != null ? getCpuLoad : detectMethod("getSystemCpuLoad");
+        processCpuUsage = detectMethod("getProcessCpuLoad");
+        currentSystemCpuUsageSupplier = () -> invoke(systemCpuUsage);
+        currentProcessCpuUsageSupplier = () -> invoke(processCpuUsage);
+    }
+
+    private final double targetCpuUsage;
+
+    private final DoubleSupplier systemCpuUsageSupplier;
+
+    private final DoubleSupplier processCpuUsageSupplier;
+
+    private final double targetProcessCpuLoad;
+
+    /**
+     * Instantiates a new Default cpu health checker.
+     *
+     * @param cpuUsageThreshold the target cpu usage
+     * @param processCpuLoadThreshold the target process cpu usage
+     */
+    private CpuHealthChecker(double cpuUsageThreshold, double processCpuLoadThreshold) {
+        this(cpuUsageThreshold, processCpuLoadThreshold,
+                currentSystemCpuUsageSupplier, currentProcessCpuUsageSupplier);
+    }
+
+    private CpuHealthChecker(double cpuUsageThreshold, double processCpuLoadThreshold,
+                             DoubleSupplier systemCpuUsageSupplier, DoubleSupplier processCpuUsageSupplier) {
+        this.targetCpuUsage = cpuUsageThreshold;
+        this.targetProcessCpuLoad = processCpuLoadThreshold;
+        this.systemCpuUsageSupplier = systemCpuUsageSupplier;
+        this.processCpuUsageSupplier = processCpuUsageSupplier;
+    }
+
+    private static double invoke(@Nullable MethodHandle mh) {
+        if (mh == null) {
+            return Double.NaN;
+        }
+
+        try {
+            return (double) mh.invoke(operatingSystemBean);
+        } catch (Throwable e) {
+            return Double.NaN;
+        }
+    }
+
+    @Nullable
+    @SuppressWarnings("ReturnValueIgnored")
+    private static MethodHandle detectMethod(final String name) {
+        if (operatingSystemBeanClass == null) {
+            return null;
+        }
+        try {
+            // ensure the Bean we have is actually an instance of the interface
+            operatingSystemBeanClass.cast(operatingSystemBean);
+            final Method method = operatingSystemBeanClass.getMethod(name);
+            return LOOKUP.unreflect(method);
+        } catch (ClassCastException | NoSuchMethodException | SecurityException | IllegalAccessException e) {
+            logger.warn(
+                    "Failed to detect method {}.{} for {}",
+                    operatingSystemBeanClass.getSimpleName(),
+                    name,
+                    CpuHealthChecker.class.getSimpleName(),
+                    e
+            );
```

Comment:
Can you reformat these lines?

```suggestion
            logger.warn("Failed to detect method {}.{} for {}", operatingSystemBeanClass.getSimpleName(),
                        name, CpuHealthChecker.class.getSimpleName(), e);
```